buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
 								{
-									BuildpackID: buildpackID,
-									LayerName:   "",
-									LayerType:   buildpack.LayerTypeLaunch,
-									Path:        filepath.Join(layersDir, buildpackID, "launch.sbom.cdx.json"),
-								},
-								{
-									BuildpackID: buildpackID,
-									LayerName:   otherLayerName,
-									LayerType:   buildpack.LayerTypeLaunch,
-									Path:        filepath.Join(layersDir, buildpackID, "some-launch-true-cache-false-layer.sbom.cdx.json"),
+									Require: buildpack.Require{
+										Name:     "some-deprecated-bp-replace-version-dep",
+										Metadata: map[string]interface{}{"version": "some-version-new"},
+									},
+									Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"}, // no api, no homepage
 								},
 								{
-									BuildpackID: buildpackID,
-									LayerName:   layerName,
-									LayerType:   buildpack.LayerTypeBuild,
-									Path:        filepath.Join(layersDir, buildpackID, "some-layer.sbom.cdx.json"),
+									Require: buildpack.Require{
+										Name:     "some-dep",
+										Metadata: map[string]interface{}{"version": "v1"},
+									},
+									Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"}, // no api, no homepage
 								},
 								{
-									BuildpackID: buildpackID,
-									LayerName:   layerName,
-									LayerType:   buildpack.LayerTypeCache,
-									Path:        filepath.Join(layersDir, buildpackID, "some-layer.sbom.cdx.json"),
+									Require: buildpack.Require{
+										Name:     "some-replace-version-dep",
+										Metadata: map[string]interface{}{"version": "some-version-new"},
+									},
+									Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"}, // no api, no homepage
 								},
-							},
-						}, br)
-					})
-
-					it("should error if there are unsupported extensions", func() {
-						buildpackID := bpTOML.Buildpack.ID
-						bpTOML.Buildpack.SBOM = []string{"application/vnd.cyclonedx+json", "application/spdx+json", "application/vnd.syft+json"}
-						layerName := "some-layer"
-
-						h.Mkdir(t,
-							filepath.Join(layersDir, buildpackID, layerName))
-						h.Mkfile(t, "[types]\n  launch = true\n  cache = false",
-							filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.toml", layerName)))
-						h.Mkfile(t, `{"key": "some-bom-content"}`,
-							filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.cdx.json", layerName)),
-							filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.spdx.json", layerName)),
-							filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.syft.json", layerName)),
-							filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.some-unknown-format.json", layerName)))
-
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertError(t, err, fmt.Sprintf("unsupported SBOM file format: '%s'", filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.some-unknown-format.json", layerName))))
-					})
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L700-L717